### PR TITLE
CDS-56920-TAS config files

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
GitHub connector support for config files in TAS

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+```mdx-code-block
+import Tabs from '@theme/Tabs';   
+import TabItem from '@theme/TabItem';
+```
+```mdx-code-block
+<Tabs>
+    <TabItem value="Harness" label="Harness" default>
+```
+1. Enter a config file identifier name.
```
Comment (RishabhGupta34) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:26]: nit. can we add some screenshots for both Harness and Github store?

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   * **Connect Through a Harness Delegate:** Harness will make all connections using the Harness Delegate. This option is used for Harness Self-Managed Enterprise Edition Overview often, but it is also used for Harness SaaS. See [Harness self-managed enterprise edition overview](/docs/self-managed-enterprise-edition/introduction/getting-started-with-self-managed-enterprise-edition/).
+    * If you select **Connect Through a Harness Delegate**, click **Continue** and then select/add the Delegate you want to use in **Delegates Setup**. See [delegate installation overview](/docs/platform/Delegates/delegate-concepts/delegate-overview).
+12. Click **Save and Continue**.
+
+Harness tests the connection. Click **Finish** once the verification is successful.
+
+The GitHub connector is listed in Connectors.
+
```
Comment (alluvamsi-harness) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:86]: can you about these expressions <+configFile.getAsString(“configFIle_ID”)>
<+configFile.getAsBase64("configFIle_ID")>

Comment (roshnisarangadharan) [reply docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:86]: I didn't understand this comment. @alluvamsi-harness

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+---
+title: Add config files for TAS
+description: Add config files for TAS by using Harness.
+sidebar_position: 2
+---
+
+Service configuration files enable you to specify information in the Harness service that can be referenced in other parts of the Harness application. In this topic we'll cover using these configuration files for Tanzu Application Service (TAS, formerly PCF).
```
Comment (brian-f-harness) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:7]:
```suggestion
Service configuration files enable you to specify information in the Harness service that can be referenced in other parts of the Harness application. In this topic, we'll cover using these configuration files for Tanzu Application Service (TAS, formerly PCF).
```

**Review by brian-f-harness — APPROVED** [review]
lgtm. Added a couple minor suggestions.

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+6. Select **Add File** to attach multiple files as a single config file.
+7. Select **Submit**.
+  
+  ![](./static/config-file-git.png)
+
+You can view the config files added to your service definition.
+
+You can reference this config file as an expression in a pipeline from the **Execution** tab. 
+
+For example, in a Tanzu Command step, you can reference the config file **Inline** by entering the following expression.
```
Comment (RishabhGupta34) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:61]: can you remove `Inline`?

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+You can view the config files added to your service definition.
+
+You can reference this config file as an expression in a pipeline from the **Execution** tab. 
+
+For example, in a Tanzu Command step, you can reference the config file **Inline** by entering the following expression. 
+
+`echo "<+configFile.getAsString("shellsec")>"` 
+
+You can also reference the config file from the **File Store** by selecting the file.
```
Comment (RishabhGupta34) [inline docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:65]: this line seems incorrect

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md (hunk 4 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:61]: What do you mean? How will a user enter the script if they don't select Inline or File Store option?

File: docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md (hunk 5 above)
Comment (roshnisarangadharan) [reply docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/add-config-files.md:65]: From script, either one should select inline or file store options, no?

**Pull request merged**